core/src/main/java/org/opensearch/sql/planner/physical/ProjectOperator.java

```diff
@@ -65,10 +65,14 @@ public ExprValue next() {
     // ParseExpression will always override NamedExpression when identifier conflicts
     // TODO needs a better implementation, see https://github.com/opensearch-project/sql/issues/458
     for (NamedExpression expr : namedParseExpressions) {
-      ExprValue value = inputValue.bindingTuples()
-          .resolve(((ParseExpression) expr.getDelegated()).getExpression());
-      if (value.isMissing()) {
-        // value will be missing after stats command, read from inputValue if it exists
+      if (projectList.stream()
+          .noneMatch(field -> field.getNameOrAlias().equals(expr.getNameOrAlias()))) {
+        continue;
+      }
```

Comment:
Which case covered, could you explain more?

Reply:
1. add all expressions in project list that are not parse expressions
2. add all parse expressions that are also in project list
     1. if parse expression is not pushed down as aggs, run parse logic in memory
     1. if parse expression is pushed down as aggs (stats), read it from inputValue